### DISCUSSION TITLE
Contract Code Coverage Reporting / Enforcement

### DISCUSSION THREAD

**Discussion by apbendi** [dcgtc/dgrants#75] [Architecture & Design] [answered]
We can generate a code coverage report for the smart contracts by running `yarn coverage` from the contracts subdir. Do we want to enforce ~100% coverage as a requirement for PR's to merged? If so, there a way to run the coverage report as part of our CI, and fail at a certain threshold? cc @wildmolasses on this point

**Comment by thelostone-mc** [comment]
If it's just contracts, I'd say we should consider it just to ensure our code is well tested and works exactly they way we intend it.
Makes auditing less stressful as well

**Comment by wildmolasses** [comment]
> If so, there a way to run the coverage report as part of our CI, and fail at a certain threshold?

for sure! if an issue results from this discussion id be happy to implement!

**Reply by mds1** [reply]
This feels like a "simple enough" task that I figured it's ok to create an issue from it, so here you go! https://github.com/dcgtc/dgrants/issues/82

**Comment by apbendi** [comment, marked as answer]
Will be implemented in #82
